manticore/core/smtlib/solver.py

```diff
@@ -773,5 +784,26 @@ def thread(solver):
             t.join()
         return result
 
+class SelectedSolver:
+    choice = None
+
+    @classmethod
+    def instance(cls):
+        if consts.solver == consts.solver.auto:
+            if cls.choice is None:
+                if os.path.exists(consts.yices_bin):
+                    cls.choice = consts.solver.yices
+                elif os.path.exists(consts.z3_bin):
+                    cls.choice = consts.solver.z3
+                elif os.path.exists(consts.cvc4_bin):
+                    cls.choice = consts.solver.cvc4
```

Comment:
I think what you're trying to do here is a bit more complicated than this. We want to use [`shutil.which`](https://docs.python.org/3.6/library/shutil.html#shutil.which) if it will just be the name of the executable. If it's a full path, then `os.path.exists` is correct.